docs/extensibility/snippets/MailDevResource/MailKit.Client/MailKitClientFactory.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Net;
+using MailKit.Net.Smtp;
+
+namespace MailKit.Client;
+
+/// <summary>
+/// A factory for creating <see cref="ISmtpClient"/> instances
+/// given a <paramref name="smtpUri"/> (and optional <paramref name="credentials"/>).
+/// </summary>
+/// <param name="settings">
+/// The <see cref="MailKitClientSettings"/> settings for the SMTP server
+/// </param>
+/// <param name="credentials">
+/// The optional <see cref="ICredentials"/> used to authenticate to the SMTP server
+/// </param>
+public sealed class MailKitClientFactory(
+    MailKitClientSettings settings,
+    ICredentials? credentials = null) : IDisposable
+{
+    private SmtpClient? _client;
+
+    /// <summary>
+    /// Gets an <see cref="ISmtpClient"/> instance in the connected state
+    /// (and that's been authenticated if configured).
+    /// </summary>
+    /// <param name="cancellationToken">Used to abort client creation and connection.</param>
+    /// <returns>A connected (and authenticated) <see cref="ISmtpClient"/> instance.</returns>
+    /// <remarks>
+    /// Since both the connection and authentication are considered expensive operations,
+    /// the <see cref="ISmtpClient"/> returned is intended to be used for the duration of a request
+    /// (registered as 'Scoped') and is automatically disposed of.
+    /// </remarks>
+    public async Task<ISmtpClient> GetSmtpClientAsync(
+        CancellationToken cancellationToken = default)
+    {
+        _client = new SmtpClient();
```

Comment:
Don't you want to return `_client` if it is not null?

Reply:
What do you mean? This is not `null`.

Reply:
What happens on the 2nd, 3rd, etc calls to `GetSmtpClientAsync()`? You create a new SmtpClient every time it is called.

Reply:
Ah, yes. Okay, I'll fix that too.